### ISSUE TITLE
Bun cannot start expo app

### ISSUE THREAD

**Issue by Nanome203** [oven-sh/bun#14481]
### What version of Bun is running?

1.1.30+7996d06b8

### What platform is your computer?

Microsoft Windows NT 10.0.22631.0 x64

### What steps can reproduce the bug?

```pwsh
bun create expo@latest my-app
cd my-app
bun --bun run start
```

### What is the expected behavior?

Metro bundler should get started and i can run my app with expo go

### What do you see instead?

```pwsh
Starting Metro Bundler
Error while reading cache, falling back to a full crawl:
37 |   getCacheFilePath() {
38 |     return this._cachePath;
39 |   }
40 |   async read() {
41 |     try {
42 |       return (0, _v.deserialize)(
                         ^
TypeError: Unable to deserialize data.
      at D:\my-app\node_modules\metro-file-map\src\cache\DiskCacheManager.js:42:21
      at read (D:\my-app\node_modules\metro-file-map\src\cache\DiskCacheManager.js:40:16)
      at D:\my-app\node_modules\metro-file-map\src\index.js:285:39
      at read (D:\my-app\node_modules\metro-file-map\src\index.js:281:16)
      at D:\my-app\node_modules\metro-file-map\src\index.js:187:36
      at D:\my-app\node_modules\metro-file-map\src\index.js:184:41
      at build (D:\my-app\node_modules\metro-file-map\src\index.js:236:9)
      at new DependencyGraph (D:\my-app\node_modules\metro\src\node-haste\DependencyGraph.js:45:34)
      at new Bundler (D:\my-app\node_modules\metro\src\Bundler.js:7:22)
      at new IncrementalBundler (D:\my-app\node_modules\metro\src\IncrementalBundler.js:24:21)
      at new Server (D:\my-app\node_modules\metro\src\Server.js:84:21)
      at D:\my-app\node_modules\metro\src\index.flow.js:45:18

› Detected a change in babel.config.js. Restart the server to see the new results. You may need to clear the bundler cache with the --clear flag for your changes to take effect.
TypeError [ERR_INVALID_ARG_TYPE]: The "streamReadable" argument must be an stream.Readable. Received an instance of Pipeline
TypeError: The "streamReadable" argument must be an stream.Readable. Received an instance of Pipeline
    at new NodeError (node:stream:443:19)
    at newReadableStreamFromStreamReadable (node:stream:2412:41)
    at new Response (node-fetch:48:13)
    at new NFCResponse (D:\my-app\node_modules\@expo\cli\build\src\api\rest\cache\response.js:21:8)
    at D:\my-app\node_modules\@expo\cli\src\api\rest\cache\wrapFetchWithCache.ts:194:7
    at processTicksAndRejections (unknown:7:39)
error: script "start" exited with code 1
```

### Additional information

It works fine with Node.js, so i'm running expo app with Node.js for now
Related issue: #13390

**Comment by Zoxive** [comment]
(This might help someone googling)

Im on a mac and got this.. and my problem was nodejs was messed up from a bad brew install.
I was getting `env: node: No such file or directory` on node
Reinstalling node and specifically the last step via https://stackoverflow.com/a/48736639/982190 fixed me.

**Comment by Nanome203** [comment]
#### Updates: 
In Bun v1.1.36, `bun --bun run start` now shows new error (not sure if previous one was fixed)

```pwsh
$ expo start
Starting project at D:\test
75 |         });
76 |         return records;
77 |     }
78 | }
79 | function createTelemetryFetch() {
80 |     const agent = new (_undici()).RetryAgent(new (_undici()).Agent(), {
                       ^
TypeError: undefined is not a constructor (evaluating 'new (_undici()).RetryAgent(new (_undici()).Agent, {
      maxRetries: 3,
      retryAfter: !0,
      minTimeout: 500,
      maxTimeout: 2000,
      timeoutFactor: 2
    })')
      at createTelemetryFetch (D:\test\node_modules\@expo\cli\build\src\utils\telemetry\clients\FetchClient.js:80:19)
      at new FetchClient (D:\test\node_modules\@expo\cli\build\src\utils\telemetry\clients\FetchClient.js:28:26)
      at createClientFromStrategy (D:\test\node_modules\@expo\cli\build\src\utils\telemetry\Telemetry.js:119:37)
      at setStrategy (D:\test\node_modules\@expo\cli\build\src\utils\telemetry\Telemetry.js:53:24)
      at recordCommand (D:\test\node_modules\@expo\cli\build\src\utils\telemetry\index.js:68:55)
      at D:\test\node_modules\@expo\cli\build\bin\cli:256:9

Bun v1.1.36 (Windows x64)
error: script "start" exited with code 1
```

**Comment by NovaBG03** [comment]
I tested this using `expo@53.0.20` and it worked without crashing. 
There was only a warning during startup. 

Here's the output I got when starting the project:
```shell
$ bun --bun expo start -
Starting project at /Users/nova/project
9 |     "*.fx.tsx",
        ^
warn: wildcard sideEffects are not supported yet, which means this package will be deoptimized
   at /Users/nova/project/node_modules/expo/package.json:9:5
Using src/app as the root directory for Expo Router.
Experimental React Compiler is enabled.
Experimental React 19 canary is enabled.
Fast resolver is enabled.
Starting Metro Bundler

› Metro waiting on exp://10.107.204.121:3001
› Scan the QR code above with Expo Go (Android) or the Camera app (iOS)

› Web is waiting on http://localhost:3001
```

os: macOS 15.5 (arm64)
shell: zsh
bun: 1.2.19
node: 20.19.3
npm: 10.8.2
yarn: not installed
pnpm: not installed
expo cli: 0.24.20
react: 19.1.0
react-native: 0.79.5

**Comment by Jarred-Sumner** [comment]
> warn: wildcard sideEffects are not supported yet, which means this package will be deoptimized
oh no that error should never be printing at runtime

**Comment by NovaBG03** [comment]
These are the errors I'm getting when running `bun --bun expo run:ios` and `bun --bun expo run:android`.
If I run the same commands without the `--bun` flag, they execute successfully.

```shell
$ bun --bun expo run:ios
⠋ Creating native directory (./ios)9 |     "*.fx.tsx",
        ^
warn: wildcard sideEffects are not supported yet, which means this package will be deoptimized
   at /Users/nova/project/node_modules/expo/package.json:9:5
✔ Created native directory
› Using react-native@catalog: instead of recommended react-native@0.79.5.
✔ Updated package.json | no changes
✔ Finished prebuild
⚠️  Something went wrong running `pod install` in the `ios` directory.
Command `pod install` failed.
└─ Cause: Invalid `Podfile` file: Couldn't parse JSON coming from `expo-modules-autolinking` command:
unexpected token at ''.

 #  from /Users/nova/project/apps/app-name/ios/Podfile:17
 #  -------------------------------------------
 #  target 'app-name' do
 >    use_expo_modules!
 #  
 #  -------------------------------------------

pod install --repo-update --ansi exited with non-zero code: 1
› Planning build
› Executing app-name » [CP] Check Pods Manifest.lock

❌  error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.



› 1 error(s), and 0 warning(s)

CommandError: Failed to build iOS project. "xcodebuild" exited with error code 65.
error: "expo" exited with code 1
```

```shell
$ bun --bun expo run:android
9 |     "*.fx.tsx",
        ^
warn: wildcard sideEffects are not supported yet, which means this package will be deoptimized
   at /Users/nova/project/node_modules/expo/package.json:9:5
✔ Created native directory
› Using react-native@catalog: instead of recommended react-native@0.79.5.
✔ Updated package.json | no changes
✔ Finished prebuild
› Building app...
Configuration on demand is an incubating feature.

FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/nova/project/apps/app-name/android/settings.gradle' line: 29

* What went wrong:
A problem occurred evaluating settings 'android'.
> Process 'command 'node'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 695ms
14 actionable tasks: 2 executed, 12 up-to-date
Error: /Users/nova/project/apps/app-name/android/gradlew app:assembleDebug -x lint -x test --configure-on-demand --build-cache -PreactNativeDevServerPort=8081 -PreactNativeArchitectures=arm64-v8a exited with non-zero code: 1
Error: /Users/nova/project/apps/app-name/android/gradlew app:assembleDebug -x lint -x test --configure-on-demand --build-cache -PreactNativeDevServerPort=8081 -PreactNativeArchitectures=arm64-v8a exited with non-zero code: 1
    at completionListener (/Users/nova/project/node_modules/@expo/spawn-async/build/spawnAsync.js:42:27)
    at emit (node:events:98:22)
    at #maybeClose (node:child_process:746:16)
    at #handleOnExit (node:child_process:520:72)
    at processTicksAndRejections (native:7:39)
    ...
    at spawnAsync (/Users/nova/project/node_modules/@expo/spawn-async/build/spawnAsync.js:7:23)
    at spawnGradleAsync (/Users/nova/project/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:107:40)
    at spawnGradleAsync (/Users/nova/project/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:101:33)
    at assembleAsync (/Users/nova/project/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:82:18)
    at assembleAsync (/Users/nova/project/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:64:30)
    at runAndroidAsync (/Users/nova/project/node_modules/@expo/cli/build/src/run/android/runAndroidAsync.js:85:27)
    at processTicksAndRejections (native:7:39)
error: "expo" exited with code 1
```

**Comment by robobun** [comment]
Same bug as #8487: Bun's node-fetch Response constructor calls Readable.toWeb() on any legacy Stream body, so the minipass Pipeline that node-fetch-cache (vendored into @expo/cli as NFCResponse) passes to new Response() fails with `The "streamReadable" argument must be an stream.Readable. Received an instance of Pipeline`. The fix for this (PR #35560) is tracked under #8487 and covers the Expo failure reported here.

Duplicate of #8487

Please add a 👍 or any extra details over there so it all stays in one place. If this is actually a different problem, say so here and we'll reopen.
